build_tools/docker/gradle-android/Dockerfile

```diff
@@ -0,0 +1,63 @@
+# Copyright 2021 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# An image for cross-compiling IREE's TFLite Java Bindings with Gradle and
+# CMake.
+
+ARG JDK_VERSION=11
+
+ARG GRADLE_VERSION=7.1.1
+ARG GRADLE_DIST=bin
+
+ARG ANDROID_SDK_VERSION=7583922
+ARG ANDROID_NDK_VERSION=21.4.7075529
+
+FROM gcr.io/iree-oss/util@sha256:40846b4aea5886af3250399d6adfdb3e1195a8b0177706bb0375e812d62dc49c AS install-deps
+ARG GRADLE_VERSION
+ARG GRADLE_DIST
+ARG ANDROID_SDK_VERSION
+
+# Download and install Gradle
+RUN cd /opt && \
+    wget -q https://services.gradle.org/distributions/gradle-${GRADLE_VERSION}-${GRADLE_DIST}.zip && \
+    unzip gradle*.zip && \
+    rm gradle*.zip && \
+    ln -s /opt/gradle-${GRADLE_VERSION}/bin/gradle /usr/bin/gradle
+
+# Download and install Android SDK
+# Note: Uses the latest SDK version from https://developer.android.com/studio,
+# however Gradle will automatically download any additional SDK/tooling versions
+# as necessary.
+ENV ANDROID_SDK_ROOT /opt/android-sdk
+RUN mkdir -p ${ANDROID_SDK_ROOT}/cmdline-tools && \
+    wget -q https://dl.google.com/android/repository/commandlinetools-linux-${ANDROID_SDK_VERSION}_latest.zip && \
+    unzip *tools*linux*.zip -d ${ANDROID_SDK_ROOT}/cmdline-tools && \
+    mv ${ANDROID_SDK_ROOT}/cmdline-tools/cmdline-tools ${ANDROID_SDK_ROOT}/cmdline-tools/tools && \
+    rm *tools*linux*.zip
+
+FROM gcr.io/iree-oss/cmake@sha256:9d9953acf5ca0cf1ff3e8de32f10f24dfab1c4e8ec5d1fc047f556024ee4bed6 as final
```

Comment:
would be nice to inherit from https://github.com/google/iree/blob/main/build_tools/docker/cmake-android/Dockerfile or just use a single Android image instead of having two different Android images

we have about 15 too many Docker images right now :P

Reply:
This PR isn't necessarily the place to decide how to consolidate our Docker images, but I still think that "building for Android" with the NDK and/or SDK could go through a single image. If we have to specially configure each image just to get a working build then the underlying build system needs to be simplified.